src/plugins/vis_type_vega/public/vega_view/vega_base_view.js

```diff
@@ -63,6 +63,7 @@ export class VegaBaseView {
     this._parser = opts.vegaParser;
     this._serviceSettings = opts.serviceSettings;
     this._filterManager = opts.filterManager;
+    this._vis = opts.vis;
```

Comment:
pass just applyFilter method in (and keep reference to it) instead of whole vis, to make it clearer what dependencies are